### PR TITLE
[stable/postgresql] Add support for the official postgres image env vars

### PR DESCRIPTION
**Pull request by tompizmor**
Signed-off-by: tompizmor <tompizmor@gmail.com>

#### What this PR does / why we need it:

~~---------------------------
       DO NOT MERGE
The bitnami/image release is still work-in-progress. I am sending the PR before release the bitnami/postgresql image so users that are interested on using the chart with the official image can start reviewing and testing it.
---------------------------~~

This PR updates the Postgres chart to configure the bitnami/postgresql image using the same env vars supported by the [docker maintained postgres image](https://hub.docker.com/_/postgres?tab=description).

This change allows users to switch from using the `bitnami/postgresql` image to use the docker maintained `postgresql` image. 
Note that all the features supported by the bitnami image are not supported by the docker library one. For example, the replication configuration. 

#### Which issue this PR fixes
This PR could be related to the follwing issues:
  - https://github.com/helm/charts/issues/10451
  - https://github.com/helm/charts/issues/8735
#### Special notes for your reviewer:
~~Do not merge yet. We need to release the bitnami/postgresql to be compatible with the official iamge environment variable. The release will probably happen next week.~~

The Bitnami PostgreSQL has been already released: https://github.com/bitnami/bitnami-docker-postgresql#9611-r66-9611-ol-7-r83-1060-r68-1060-ol-7-r83-1110-r62-and-1110-ol-7-r79

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md

### CODE REVIEW

**Comment by tompizmor** [comment]
cc/ @arianitu @marcoceppi @desaintmartin @prydonius @carrodher

**Comment by marcoceppi** [comment]
I've been working through testing this change, so far my initial impressions are :+1: I'll report back when I've completed my trials

File: stable/postgresql/README.md [hunk 1]
```diff
@@ -45,88 +45,91 @@ The command removes all the Kubernetes components associated with the chart and
...
+| `existingSecret`                              | Name of existing secret to use for PostgreSQL passwords                                                                | `nil`                                                         |
+| `postgresqlUsername`                          | PostgreSQL admin user                                                                                                  | `postgres`                                                    |
+| `postgresqlPassword`                          | PostgreSQL admin password                                                                                              | _random 10 character alphanumeric string_                     |
+| `postgresqlDatabase`                          | PostgreSQL database                                                                                                    | `nil`                                                         |
+| `postgresqlDataDir`                           | PostgreSQL data dir folder                                                                                             | `/bitnami/postgresql` (same value than persistence.mountPath) |
```
Comment (vikram-bitnami) [inline stable/postgresql/README.md]: same value as

File: stable/postgresql/README.md [hunk 2]
```diff
@@ -212,6 +215,25 @@ With NetworkPolicy enabled, traffic will be limited to just port 5432.
 For more precise policy, set `networkPolicy.allowExternal=false`. This will only allow pods with the generated client label to connect to PostgreSQL.
 This label will be displayed in the output of a successful install.
 
+## Deploy chart using Docker Official PostgreSQL image
```
Comment (vikram-bitnami) [inline stable/postgresql/README.md]: image -> Image (capitalized as per the official docs https://docs.docker.com/docker-hub/official_images/)

File: stable/postgresql/README.md [hunk 3]
```diff
@@ -212,6 +215,25 @@ With NetworkPolicy enabled, traffic will be limited to just port 5432.
...
+
+From chart version 4.0.0, it is possible to use the stable/postgresql chart with the Docker Official PostgreSQL image.
+Besides specifying the new docker repository and tag, it is important to modify the PostgreSQL data dir and volume mount point. Basically, the PostgreSQL data dir cannot be the mount point directly, it has to be a subdirectory.
```
Comment (vikram-bitnami) [inline stable/postgresql/README.md]: docker -> Docker

dir -> directory

File: stable/postgresql/README.md [hunk 4]
```diff
@@ -212,6 +215,25 @@ With NetworkPolicy enabled, traffic will be limited to just port 5432.
 For more precise policy, set `networkPolicy.allowExternal=false`. This will only allow pods with the generated client label to connect to PostgreSQL.
 This label will be displayed in the output of a successful install.
 
+## Deploy chart using Docker Official PostgreSQL image
+
+From chart version 4.0.0, it is possible to use the stable/postgresql chart with the Docker Official PostgreSQL image.
```
Comment (vikram-bitnami) [inline stable/postgresql/README.md]: it is possible to use the stable/postgresql chart -> this sounds a bit odd, shouldn't we say "it is possible to use this chart"? Or do we have multiple tags/versions for our chart, so that's why we need to name the tag here? Do we want to add a link to this tag?

image -> Image

Do you want to link to the image page on Docker Hub?

File: stable/postgresql/README.md [hunk 5]
```diff
@@ -212,6 +215,25 @@ With NetworkPolicy enabled, traffic will be limited to just port 5432.
...
+             --set image.repository=postgres \
+             --set image.tag=10.6 \
+             --set postgresqlDataDir=/data/pgdata \
+             --set persistence.mountPath=/data/ \
+             stable/postgresql
+```
+
+## Differences between Bitnami PostgreSQL image and [Docker Official](https://hub.docker.com/_/postgres) image
+
+- The Docker Official PostgreSQL image does not support replication. If you pass any replication environment variable, this would be ignored. The only environment variables supported by the Docker Official image are POSTGRES_USER, POSTGRES_DB, POSTGRES_PASSWORD, POSTGRES_INITDB_ARGS, POSTGRES_INITDB_WALDIR and PGDATA. All the remaining environment variables are additions to the Bitnami PostgreSQL image.
```
Comment (vikram-bitnami) [inline stable/postgresql/README.md]: additions to -> specific to

File: stable/postgresql/README.md [hunk 6]
```diff
@@ -212,6 +215,25 @@ With NetworkPolicy enabled, traffic will be limited to just port 5432.
...
+- The Bitnami PostgreSQL image is non-root by default. This requires that you run the pod with securityContext and updates the permissions of the volume with at initContainer. As a reward for the extra configuration, the pod follows security best practices and is prepared to run on Kubernetes distributions with hard security constraints like OpenShift.
```
Comment (vikram-bitnami) [inline stable/postgresql/README.md]: the volume with at initContainer -> is there a typo here

Do you want to put code-related things like securityContext, initContainer in code font?

As a reward for the extra configuration... -> I would reword this, maybe instead say "a key benefit of this configuration is that the pod follows..."

**Comment by desaintmartin** [comment]
Hello,
As usual, a "how to upgrade to 4.0.0" is welcome in the readme! ;)

**Comment by tompizmor** [comment]
@vikram-bitnami thanks for the README review. I just made the changes

@desaintmartin I agree as long as I need to update the major version of the change. However, I am not really sure if I should increase the major version because of this change. 
I think this is not a breaking change as I haven't renamed any value parameter but the name of the environment variable (which should be a transparent change for users). Apart from that, I have added a couple of new parameters but that should only require a minor version bump.
If anyone has a different opinion please let me know!

**Comment by desaintmartin** [comment]
Oh, nice, actually if it does not break something, it is not a breaking change thus we should only increase a minor version, right?

**Comment by tompizmor** [comment]
Right! At the beginning, I thought it was a breaking change but after my tests, I didn't have troubles updating it.

BTW, the Bitnami PostgreSQL image was released so this changes should also work when using the Bitnami image. 
@marcoceppi did you find something we need to address in your tests?

**Comment by tompizmor** [comment]
@marcoceppi @arianitu Let me know if you have any comments! I would like to get this PR merged during this week

**Review by carrodher — APPROVED** [review]
Thanks @tompizmor
